simulator/resourcewatcher/resourcewatcher.go

```diff
@@ -0,0 +1,323 @@
+type StreamWriter interface {
+	Write(we *sw.WatchEvent) error
+}
+
+// Service watches simulator's resources.
+type Service struct {
+	client clientset.Interface
+}
+
+// NewService initializes Service.
+func NewService(client clientset.Interface) *Service {
+	return &Service{
+		client: client,
+	}
+}
+
+// Watch watches each simulator's resources and send notified events to the frontend continuously.
+func (s *Service) Watch(ctx context.Context, stream sw.ResponseStream, lrVersions *LastResourceVersions) error {
+	sw := sw.NewStreamWriter(stream)
+	proxies := []*resourceEventProxy{
+		newresourceEventProxy(sw, s.client.CoreV1().RESTClient(), Pods, &corev1.Pod{}, lrVersions.Pods),
+		newresourceEventProxy(sw, s.client.CoreV1().RESTClient(), Nodes, &corev1.Node{}, lrVersions.Nodes),
+		newresourceEventProxy(sw, s.client.CoreV1().RESTClient(), Pvs, &corev1.PersistentVolume{}, lrVersions.Pvs),
+		newresourceEventProxy(sw, s.client.CoreV1().RESTClient(), Pvcs, &corev1.PersistentVolumeClaim{}, lrVersions.Pvcs),
+		newresourceEventProxy(sw, s.client.StorageV1().RESTClient(), Scs, &storagev1.StorageClass{}, lrVersions.Scs),
+		newresourceEventProxy(sw, s.client.SchedulingV1().RESTClient(), Pcs, &schedulingv1.PriorityClass{}, lrVersions.Pcs),
+	}
+	runctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	for _, p := range proxies {
+		go s.Run(p, runctx.Done(), cancel)
+	}
+
+	select {
+	case <-runctx.Done():
+		// runctx monitors s.Run (ListAndWatch) for each resource.
+		// If some error occurs in the process before starting the watch,
+		// this error is returned.
+		return xerrors.Errorf("failed to run ListAndWatch: %w", runctx.Err())
+	case <-ctx.Done():
+		// This method will return an error and finish to event send when the connection from a client is closed.
+		// It includes browser reload, ReadableStream.cancel() calling and so on.
+		// This is to allow the front end to handle stream connection disconnections.
+		return nil
+	}
+}
+
+// Run runs ListAndWatch method.
+// If an error is returned, call cancel to abort ListAndWatch of other resources being processed in parallel.
+func (s *Service) Run(p *resourceEventProxy, stopCh <-chan struct{}, cancel context.CancelFunc) {
+	defer cancel()
+	// ListAndWatch usually continues to wait for WATCH to end and does not return any value.
+	if err := s.ListAndWatch(p, stopCh); err != nil {
+		cancel()
+		klog.Errorf("call ListAndWatch: %w", err)
+	}
+}
+
+// ListAndWatch runs list and watch on the target resource. The list is not always ran
+// This method returns error unless an error occurs in the watch. If an error occurs in the watch,
+// it outputs a log and re-run the watch.
+func (s *Service) ListAndWatch(p ResourceEventProxy, stopCh <-chan struct{}) error {
+	lw := createListWatch(p)
+	// If the lastResourceVersion isn't specified by client, call the list and return the result as ADDED event first.
+	if p.LastResourceVersion() == "" {
```

Comment:
If a client will not set the `lastresourceversion`, the value will be `""`.
And then the server calls `list` method of the resource before starting the `watch`.